source/encoding.c

```diff
@@ -17,6 +17,48 @@
 
 #include <assert.h>
 #include <ctype.h>
+#include <stdlib.h>
+
+#ifdef USE_SIMD_ENCODING
+size_t aws_common_private_base64_decode_sse41(const unsigned char *in, unsigned char *out, size_t len);
+size_t aws_common_private_base64_encode_sse41(const unsigned char *in, unsigned char *out, size_t len);
+bool aws_common_private_has_b64_simd();
+#else
+size_t aws_common_private_base64_decode_sse41(const unsigned char *in, unsigned char *out, size_t len) {
+    (void)in;
+    (void)out;
+    (void)len;
+
+    /*
+     * This volatile variable prevents MSVC from flagging code after the call to this stub as unreachable
+     */
+
+    volatile bool do_abort = true;
+    if (do_abort) {
+        abort();
+    }
```

Comment:
So this function always aborts. Why does its implementation even exist? Shouldn't it just not be invoked on platforms that don't support it?

Reply:
Correct. This is avoiding putting preprocessor blocks inside of the aws_base64_encode/decode functions, and instead consolidating them in one place at the top of the file.

Reply:
Ok. How about using a pragma instead of this somewhat strange construct? Something like:
```
#ifdef _MSC_VER
#pragma warning(disable:<WHATEVER THE WARNING NUMBER>)
#endif
```
This will 1) be a fix specific to the misbehaving compiler and 2) not break in surprising ways when the Visual Studio developers change their way of detecting dead code.

Reply:
`(void)x` is a very standard way of suppressing unused variable warnings, but I'll suppress it with a pragma here.

Reply:
`(void)x` is, but this construct using `volatile` seems less standard, but I could be wrong.

Reply:
Actually, this might warn on non-x86 GCC as well, so I'll stick with (void)s.

Reply:
The issue that triggered the warning was a visible call to abort() - MSVC then determined that the function containing this unconditional abort() never returned, and created an unreachable code warning in the caller. The fix is to just return `-1` (a failure indication) instead; since the function is unreachable in the first place, we don't care too much what it returns.